internal/am/poll_transfer.go

```diff
@@ -52,17 +82,33 @@ func (a *PollTransferActivity) Execute(ctx context.Context, params *PollTransfer
 			return nil, ctx.Err()
 		case <-ticker.C:
 			reqCtx, cancel := context.WithTimeout(ctx, a.cfg.PollInterval/2)
-			resp, httpResp, err := a.poll(reqCtx, params.TransferID, cancel)
-			if err == errWorkOngoing {
+			resp, _, err := a.poll(reqCtx, params.TransferID, cancel)
+			if err == ErrWorkOngoing {
+				// Update preservation task data.
+				count, err := pa.SaveTasks(ctx, params.TransferID)
```

Comment:
Should we use the context with timeout previously defined to save the tasks? I'm thinking if things go wrong here we could be missing a heartbeat, but I don't know if that's a big deal.